packages/devtools-evm-hardhat/src/simulation/compose.ts

```diff
@@ -0,0 +1,93 @@
+import { pipe } from 'fp-ts/lib/function'
+import * as RR from 'fp-ts/ReadonlyRecord'
+import type { ComposeSpec, ComposeSpecService } from '@layerzerolabs/devtools'
+import { type AnvilOptions, createAnvilCliOptions } from '@layerzerolabs/devtools-evm'
+import type { ComposeSpecServices } from '@layerzerolabs/devtools'
+import type { SimulationConfig } from './types'
+
+/**
+ * Creates a docker compose service specification for an anvil-based EVM node
+ *
+ * @param {AnvilOptions} anvilOptions
+ * @returns {ComposeSpecService}
+ */
+export const createEvmNodeServiceSpec = (anvilOptions: AnvilOptions): ComposeSpecService => ({
+    // This service references a Dockerfile that is copied
+    // next to the resulting docker-compose.yaml
+    //
+    // The source for this Dockerfile is located in src/simulation/assets/Dockerfile.conf
+    build: {
+        dockerfile: 'Dockerfile',
+        target: 'node-evm',
+    },
+    command: ['anvil', ...createAnvilCliOptions(anvilOptions)],
+})
+
+/**
+ * Creates a docker compose service specification for an nginx-based proxy service
+ * that proxies requests to underlying EVM nodes (or their RPC URLs to be mor precise)
+ *
+ * @param {number} port
+ * @param {ComposeSpecServices} networkServices
+ * @returns {ComposeSpecService}
+ */
+export const createEvmNodeProxyServiceSpec = (
+    port: number,
+    networkServices: ComposeSpecServices
+): ComposeSpecService => ({
+    // This service references a Dockerfile that is copied
+    // next to the resulting docker-compose.yaml
+    //
+    // The source for this Dockerfile is located in src/simulation/assets/Dockerfile.conf
+    build: {
+        dockerfile: 'Dockerfile',
+        target: 'proxy-evm',
+    },
+    // This service will expose its internal 8545 port to a host port
+    //
+    // The internal 8545 port is hardcoded both here and in the nginx.conf file,
+    // the source for which is located in src/simulation/assets/nginx.conf
+    ports: [`${port}:8545`],
+    depends_on: pipe(
+        networkServices,
+        // This service will depends on the RPCs to be healthy
```

Comment:
`depends` -> `depend`

Reply:
Removed that `'s'` and replaced it with a `''` using a regular expression service running in docker for portability reasons, to save time next time I need to do this.

It's called productivity, look it up.

Reply:
I asked ChatGPT to create a script to look it up for me, then pipe it into text to speech so I can learn it while still reviewing CRs.

Reply:
Productivity stand-off happening right here right now